BedrockTimeoutCommandQueue.cpp

```diff
@@ -0,0 +1,58 @@
+#include <BedrockTimeoutCommandQueue.h>
+
+const BedrockCommand& BedrockTimeoutCommandQueue::front() const {
+    lock_guard<decltype(_queueMutex)> lock(_queueMutex);
+    if (_queue.empty()) {
+        throw out_of_range("No commands");
+    }
+
+    // has anything timed out?
+    if (_timeoutMap.begin()->first < STimeNow()) {
```

Comment:
couldn't you skip this time check by checking if the iterator is equal to the end of the map? seems like the time check would be more expensive, but not sure that it matters.

Reply:
The timeout map contains all commands, not just timed out commands, they're just sorted by timeout. We need to see if the one in front has timed out or not, it might not (probably won't) have.

Reply:
ah thanks, makes sense 👍